HIRS_Utils/src/main/java/hirs/data/persist/certificate/Certificate.java

```diff
@@ -547,13 +806,83 @@ public AttributeCertificate getAttributeCertificate() throws IOException {
         return attCertificate;
     }
 
+    /**
+     * Getter for the Authority Info Access List.
+     * @return return a list of Info Access
+     */
+    public String getAuthInfoAccess() {
+        return authorityInfoAccess;
+    }
+
+    /**
+     * Getter for the authorityKeyIdentifier.
+     * @return the ID String
+     */
+    public String getAuthKeyId() {
+        return authorityKeyIdentifier;
+    }
+
+    /**
+     * Getter for the policy statement.
+     * @return cloned bit representation of constraints
+     */
+    public byte[] getPolicyConstraints() {
+        if (policyConstraints != null) {
+            return policyConstraints.clone();
+        }
+
+        return new byte[0];
+    }
+
+    /**
+     * Getter for the keyUsage info.
+     * @return key usages
+     */
+    public String getKeyUsage() {
+        return keyUsage;
+    }
+
+    /**
+     * Getter for the extended key info.
+     * @return extended key usages
+     */
+    public String getExtendedKeyUsage() {
+        return extendedKeyUsage;
+    }
+
+    /**
+     * Getter for the algorithm.
+     * @return algorithm for the signature
+     */
+    public String getSignatureAlgorithm() {
+        return signatureAlgorithm;
+    }
+
+    /**
+     * Getter for the public key algorithm.
+     * @return  public key algorithm
+     */
+    public String getPublicKeyAlgorithm() {
+        return publicKeyAlgorithm;
+    }
+
     /**
      * @return this certificate's serial number
      */
     public BigInteger getSerialNumber() {
         return serialNumber;
     }
 
+    /**
+     * @return this Authority's Key ID serial number.
+     */
+    public BigInteger getAuthoritySerialNumber() {
+        if (this.authoritySerialNumber == null) {
+            return BigInteger.ZERO;
+        }
+        return authoritySerialNumber;
```

Comment:
At the end of the main constructor, around line 427, the bouncycastle class AuthorityKeyIdentifier method getAuthorityCertSerialNumber can return null.

Reply:
AuthorithySerialNumber can possibly be set to null around line 427 from the bouncy castle class method authKeyIdentifier.getAuthorityCertSerialNumber().

Reply:
Alrighty, if that's the case then instead we could remove the `else` block from where `authKeyIdentifier.getAuthorityCertSerialNumber()` gets called in the constructor in favor of having a 2nd `if` statement right after `if (authKeyIdentifier != null)` where `if (this.authoritySerialNumber == null)` then do `this.authoritySerialNumber == BigInteger.Zero`.

Reply:
What is the purpose of doing this? authoritySerialNumber is final and has to be set.  

Checking the status of authoritySerialNumber to be null when it is supposed to be set first causes errors.  This puts in 2 separate instances of setting a final variable.  1) authorityKeyIdentifier is not null but then sets authoritySerialNumber to null and 2) then if it is null we set the final variable again to a default value.

Reply:
Well aside from having the means to work around those errors via a Temp variable, my point here is that the default value of `authoritySerialNumber` is clearly `BigInteger.Zero` that much is made clear by it receiving that value in the default, empty constructor of `Certificate` and here again when the null check is made in the Getter.

Since the intent is for all Certificates to have/return a non-null `authoritySerialNumber` when someone goes to get it from them, then why have the middle-man null check in the getter method that gets fired off on every call to `getSerialNumber()`, and not just take care of it the one-time, up front in the Constructor and then forget about it ever having the possibility of being `null`.

Reply:
So then your preference is to not have any Getter include any form of an if check?  and just a one line return statement?

Reply:
Yep if it can be avoided.